### PR TITLE
attempt to fix failing benchmark tests

### PR DESCRIPTION
**Pull request by housengw**
Attempts to fix failing benchmark tests in #777, which has the following error:

```bash
[2022-01-14 23:15:38,591][/home/runner/work/lingua-franca/lingua-franca/bin/lfc][INFO] - 
Unable to import file:/home/runner/work/lingua-franca/lingua-franca/multirun/2022-01-14/23-14-33/43/BenchmarkRunner.lf: 
java.io.FileNotFoundException: 
/home/runner/work/lingua-franca/lingua-franca/multirun/2022-01-14/23-14-33/43/BenchmarkRunner.lf 
(No such file or directory)
```

### CODE REVIEW

File: .github/workflows/benchmark-tests.yml [hunk 1]
```diff
@@ -21,6 +21,12 @@ jobs:
       - uses: actions/checkout@v2
```
Comment (lhstrh) [inline .github/workflows/benchmark-tests.yml:21]: This action already checks out the `lingua-franca` repo (assuming that the workflow is invoked from the `lingua-franca` repo and not from elsewhere.

Comment (housengw) [reply .github/workflows/benchmark-tests.yml]: you're right

**Comment by housengw** [comment]
I fixed all the FileNotFound exceptions but now there is another error:
```bash
Error executing job with overrides: ['test_mode=True', 'iterations=1', 'benchmark=savina_concurrency_bndbuffer', 'target=lf-c', 'iterations=1']
2855
An error occurred during Hydra's exception formatting:
2856
TypeError("print_exception() got an unexpected keyword argument 'etype'")
2857
Traceback (most recent call last):
2858
  File "/opt/hostedtoolcache/Python/3.10.1/x64/lib/python3.10/site-packages/hydra/_internal/utils.py", line 283, in run_and_report
2859
    print_exception(etype=None, value=ex, tb=final_tb)  # type: ignore
2860
TypeError: print_exception() got an unexpected keyword argument 'etype'
2861

2862
During handling of the above exception, another exception occurred:
2863

2864
Traceback (most recent call last):
2865
  File "/home/runner/work/lingua-franca/lingua-franca/benchmark/runner/run_benchmark.py", line 194, in <module>
2866
    main()
2867
  File "/opt/hostedtoolcache/Python/3.10.1/x64/lib/python3.10/site-packages/hydra/main.py", line 48, in decorated_main
2868
    _run_hydra(
2869
  File "/opt/hostedtoolcache/Python/3.10.1/x64/lib/python3.10/site-packages/hydra/_internal/utils.py", line 385, in _run_hydra
2870
    run_and_report(
2871
  File "/opt/hostedtoolcache/Python/3.10.1/x64/lib/python3.10/site-packages/hydra/_internal/utils.py", line 294, in run_and_report
2872
    raise ex
2873
  File "/opt/hostedtoolcache/Python/3.10.1/x64/lib/python3.10/site-packages/hydra/_internal/utils.py", line 211, in run_and_report
2874
    return func()
2875
  File "/opt/hostedtoolcache/Python/3.10.1/x64/lib/python3.10/site-packages/hydra/_internal/utils.py", line 386, in <lambda>
2876
    lambda: hydra.multirun(
2877
  File "/opt/hostedtoolcache/Python/3.10.1/x64/lib/python3.10/site-packages/hydra/_internal/hydra.py", line 140, in multirun
2878
    ret = sweeper.sweep(arguments=task_overrides)
2879
  File "/opt/hostedtoolcache/Python/3.10.1/x64/lib/python3.10/site-packages/hydra/_internal/core_plugins/basic_sweeper.py", line 161, in sweep
2880
    _ = r.return_value
2881
  File "/opt/hostedtoolcache/Python/3.10.1/x64/lib/python3.10/site-packages/hydra/core/utils.py", line 233, in return_value
2882
    raise self._return_value
2883
  File "/opt/hostedtoolcache/Python/3.10.1/x64/lib/python3.10/site-packages/hydra/core/utils.py", line 160, in run_job
2884
    ret.return_value = task_function(task_cfg)
2885
  File "/home/runner/work/lingua-franca/lingua-franca/benchmark/runner/run_benchmark.py", line 75, in main
2886
    check_return_code(code, continue_on_error)
2887
  File "/home/runner/work/lingua-franca/lingua-franca/benchmark/runner/run_benchmark.py", line 107, in check_return_code
2888
    raise RuntimeError(
2889
RuntimeError: Command returned with non-zero exit code (4)
2890
Error: Process completed with exit code 1.
```

**Comment by mattchorlian** [comment]
I'm not sure this will fix it but there was a missing semicolon in BoundedBuffer.  It's a bit odd that the compiler didn't report this...

**Comment by housengw** [comment]
I canceled the workflow because the bounded buffer benchmark still errors on my machine with an error message that indicates `numIterations not found`. I have added the parameter in the savina files. Let's see if that fixes the issue.

**Comment by cmnrd** [comment]
Could you provide some context? As far as I know benchmark tests work on master. So what exactly are you trying to fix?

> I'm not sure this will fix it but there was a missing semicolon in BoundedBuffer. It's a bit odd that the compiler didn't report this...

Semicolons are optional in LF, they can be omitted without problem. It is purely stylistic choice whether to include semicolons or not.

**Comment by mattchorlian** [comment]
> Could you provide some context? As far as I know benchmark tests work on master. So what exactly are you trying to fix?
> 

Here is the situation as I understand it:  Arthur and I have been trying to wrap up [this PR] (https://github.com/lf-lang/lingua-franca/pull/777) and we have both been able to compile and run each benchmark as we pushed them.  However, the benchmarks were failing the CI c-benchmark-tests because the BenchmarkRunner file couldn't be found.  @housengw opened this PR to fix that issue and I believe he has done that successfully.  So, this PR can probably be merged, and the remaining error with BoundedBuffer can be absorbed into our PR and sorted out there.

File: benchmark/runner/conf/benchmark/savina_concurrency_barber.yaml [hunk 2]
```diff
@@ -5,6 +5,7 @@ params:
   waiting_room_size: 1000
   production_rate: 1000
   haircut_rate: 1000
+  num_iterations: 12
```
Comment (cmnrd) [inline benchmark/runner/conf/benchmark/savina_concurrency_barber.yaml:8]: The list here is intended for benchmark specific parameters. `num_iterations` and `threads` are global parameters that apply to all benchmark. Therefore, `num_iterations` should not be listed here and also not be passed to the `gen_args`. This is taken care of automatically.

Note that this comment applies to all the changed yaml files.

**Pull request merged**